packages/client/src/data_commons_client.ts

```diff
@@ -372,9 +391,52 @@ class DataCommonsClient {
       populationObservations
     );
 
+    // Filter by start/end date if specified
```

Comment:
What should happen if the observation date is different granularity than the params?

case 1:
observationDate: 2023-01-06
startDate: 2023

case 2:
observationDate: 2023
startDate: 2023-01-06

Reply:
Hmm looks like the logic here deviates from the logic in line_tile.tsx
`data_commons_client.ts` -> lexicographically comparison
`line_tile.tsx` -> truncates startDate/endDate to the length of the observation date, then does a lexicographical comparison

I think the current line_tile.tsx behavior isn't quite right-- consider this example:

```
  <datacommons-line
  header="US Population in May 2010"
  places="country/USA"
  variables="Count_Person"
  colors="blue"
  startDate="2010-05-05"
  endDate="2010-05-06"
></datacommons-line>
```

Shows a chart with an observation from "2010":

<img width="977" alt="Screenshot 2024-04-03 at 3 15 25 PM" src="https://github.com/datacommonsorg/website/assets/13766/2975d56e-5cef-47e4-aacb-db62cbcbb87f">

But, i would expect it to only show observations strictly between 2010-05-05 and 2010-05-06, and thus no data to be shown

Open question:
- if startDate = 2014, and endDate=2015, does this mean only include observations for 2014, or include observations for 2014 and 2015?

I vote we do either 
(1) (preference) strict lexicographical comparison where startDate <= obs <= endDate , as i think this is easy enough to explain to users.
(2) instead of truncating the start/end dates to match the observation's length, truncate the observation's length to match the start/end dates length, and run the final calculation as startDate <= truncated(obs) <= endDate. this would mean startDate=2014 and endDate=2015 would show observations from any time in 2014 or 2015, which seems slightly off to me (i prefer it would just show 2014 or observations from exactly 2015, but open to other ideas)

Reply:
hm, good points. I think:
- if startDate = 2014 and endDate=2015, we should include observations for both 2014 and 2015 because as per the comment, endDate is "latest date to show on the chart" https://github.com/datacommonsorg/website/blob/ead9e365979eb0e4f66f4981b8103e55b7a2359a/static/js/components/tiles/line_tile.tsx#L94-L97
- & since I think we should include observations from 2015, I prefer (2)

Reply:
Done